app/src/DataTable.tsx

```diff
@@ -0,0 +1,103 @@
+import React from "react";
+import { useTable, usePagination, useSortBy } from "react-table";
+import { Cell, HeaderGroup, Row } from "react-table";
+import MaUTable from "@material-ui/core/Table";
+import TableBody from "@material-ui/core/TableBody";
+import TableCell from "@material-ui/core/TableCell";
+import TableHead from "@material-ui/core/TableHead";
+import TableRow from "@material-ui/core/TableRow";
+import styles from "./DataTable.module.scss";
+import TableFooter from "@material-ui/core/TableFooter";
+import TablePagination from "@material-ui/core/TablePagination";
+import TableSortLabel from "@material-ui/core/TableSortLabel";
+import Paper from "@material-ui/core/Paper";
+
+function renderHeader(column: HeaderGroup<any>) {
```

Comment:
@aaronsn as these are components you could also name them as such, e.g. `TableHeader`, `TableHeaderGroup`, `TableCell`, etc. and then invoke them as JSX like:

```jsx
{row.cells.map((cell) => <TableCell {...{ cell }} /> )}
```

up to you as whether or not to keep them in the same file. There's also the concept of "render props" where you can pass these types of functions to a component as props, but I typically don't use that approach.

Reply:
converted to make them components, though I will say I think component syntax just for helper methods is a bit of a pain. It means you need to destructure all of your arguments. But I think this is more idiomatic for react so I went ahead and did it.

I'll keep them in the same file, at this point they're more of an implementation detail of the table, I just split them out because I think it makes the main Table's render method more readable. If these grow to become complex components that can stand alone we can split them out to different files.